lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -1119,6 +1132,47 @@ def get_docker_environment(env, env_files):
     return sorted(env_list)
 
 
+def get_docker_networks(networks, network_ids):
+    """
+    Validate a list of network names or a list of network dictionaries.
+    Network names will be resolved to ids by using the network_ids mapping.
+    """
+    if networks is None:
+        return None
+    parsed_networks = []
+    for network in networks:
+        if isinstance(network, string_types):
+            parsed_network = {'name': network}
+        elif isinstance(network, dict):
+            if 'name' not in network:
+                raise TypeError(
+                    '"name" is required when networks are passed as dictionaries.'
+                )
+            parsed_network = {'name': network['name']}
+            if 'aliases' in network:
+                if not all(
+                    isinstance(alias, string_types) for alias in network['aliases']
+                ):
+                    raise TypeError('Only strings are allowed as network aliases.')
+                parsed_network['aliases'] = network['aliases']
+            if 'options' in network:
+                options = network['options']
+                if not isinstance(options, dict):
+                    raise TypeError('Only dict is allowed as network options.')
+                parsed_network['options'] = network['options']
```

Comment:
```suggestion
                parsed_network['options'] = clean_dict_booleans_for_docker_api(network['options'])
```
While this also accepts `True` and `False` (and converts them to `"true"` and `"false"`), it makes sure that all keys and values are strings.